homeassistant/components/meteo_france/__init__.py

```diff
@@ -6,62 +6,61 @@
 from vigilancemeteo import VigilanceMeteoError, VigilanceMeteoFranceProxy
 import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_MONITORED_CONDITIONS
 import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 from homeassistant.util import Throttle
 
-from .const import CONF_CITY, DATA_METEO_FRANCE, DOMAIN, SENSOR_TYPES
+from .const import CONF_CITY, DOMAIN, SENSOR_TYPES
 
 _LOGGER = logging.getLogger(__name__)
 
 SCAN_INTERVAL = datetime.timedelta(minutes=5)
 
 
-def has_all_unique_cities(value):
-    """Validate that all cities are unique."""
-    cities = [location[CONF_CITY] for location in value]
-    vol.Schema(vol.Unique())(cities)
-    return value
-
+CITY_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_CITY): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS): vol.All(
```

Comment:
Not really because there is an other API (`VigilanceMeteoFranceProxy`) called when `"weather_alert" in entry.data[CONF_MONITORED_CONDITIONS]`.

But I can remove the monitored conditions and add a new boolean parameter "weather_alert".

What do you think ?

Reply:
But is it a problem to always add the weather alert entities? We only need an option of there's some problem with adding the entity for all users.

Reply:
No problem to me.

All monitored condition are fetched by `meteofranceClient` API, except weather_alert with is fetched by `VigilanceMeteoFranceProxy` API.

As I understand by reading ADR-03 is : 
- if one API call for all sensors --> no monitored conditions
- if multiple API calls --> let the user know what he wants (monitored conditions or an other parameter)

That's why I was asking fo a parameter, but maybe I'm wrong, since I donc't get everything, see below.

By reading ADR-03 : 

> Integrations should expose all available data to the backend if that data is fetched in a single API request.

Perhaps, I do not understand this sentence : 

> Integrations should only include selector logic if it make sense in the context of interface, like if it would require extra requests

Reply:
If there's a problem with the extra requests we should add an option. Eg if there are hundreds of entity types and each type requires an additional request, and doing hundreds of requests slows the integration down or imposes rate limiting.

If we're only talking about one extra request per entry, I'm thinking this might not be a problem. But I don't know this integration. That's my question.

Reply:
2 requests at the init
1 request per updates.

Removing monitored conditions 👍